src/liballoc/collections/btree/map.rs

```diff
@@ -135,7 +135,7 @@ unsafe impl<#[may_dangle] K, #[may_dangle] V> Drop for BTreeMap<K, V> {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<K: Clone, V: Clone> Clone for BTreeMap<K, V> {
+impl<K: Clone + Ord, V: Clone> Clone for BTreeMap<K, V> {
```

Comment:
You should be able to use specialization with an internal trait to avoid changing the bound here.